pkgs/development/compilers/zig/default.nix

```diff
@@ -21,6 +22,37 @@ stdenv.mkDerivation rec {
   # Fix index out of bounds reading RPATH (cherry-picked from 0.10-dev)
   patches = [ ./rpath.patch ];
 
+  # Fix build on macOS 13
+  # Note that we can't use fetchpatch, see: https://github.com/NixOS/nixpkgs/issues/32084
+  patches' = [
+    (fetchurl {
+      url = "https://github.com/ziglang/zig/commit/5b813f1a2acdc1668e39008b02a234f3da724552.patch";
+      hash = "sha256-eqaUJTE04QonxeEAKzLGyz3ak7V1YC9u29kpj5L2INM=";
+    })
+    (fetchurl {
+      url = "https://github.com/ziglang/zig/commit/e3274067f8add7db239a49cb061eaac28008cdec.patch";
+      hash = "sha256-WuISk7RZrzEo0jhbqfqL/FoKiZzscfGcoDAEzy6ik8A=";
+    })
+    (fetchurl {
+      url = "https://github.com/ziglang/zig/commit/98285b17b3887de37b630da66f09a44f42ddbe01.patch";
+      hash = "sha256-ybJNNQSAq1v6v1ZiSB0wE2vsa1u7AzcRXnhWlWx+sso=";
+    })
+    (fetchurl {
+      url = "https://github.com/ziglang/zig/commit/43dbebdd8dbccd00bd50c2df3d9d7ef8bfcddb36.patch";
+      hash = "sha256-weVoZvrxDALSNkawVyBeCo9MyApkT47R0f36PVYYbuI=";
+    })
+    (fetchurl {
+      url = "https://github.com/ziglang/zig/commit/304eb54169ad9b6d7fc278807df39ef68e76202f.patch";
+      hash = "sha256-PwxyDDUB2P0GYYzASHXVDATx+pydkIpSQQxOomNB+h4=";
+    })
+  ];
+
+  postPatch = ''
+    for patch in ${toString patches'}; do
+      patch -p1 < $patch || true
```

Comment:
Why do you use `patch -p1` manually? Can't you just extend the list of `patches` with the calls to `fetchurl`?

Reply:
Unlike #208826, it would be impossible to save the patches due to their size (for example, 1.7MB for the second patch). A better solution has to be found, for example, replace the files with those from the 0.10.0 tarball. Making this a draft.

Reply:
You could use `fetchurl` but fetch from archive.org, so the patch never changes. Does that work?

Reply:
I rewrote the derivation, so no more heavy patches involved.